specification/orbital/resource-manager/Microsoft.Orbital/stable/2022-11-01/orbital.json

```diff
@@ -0,0 +1,2280 @@
+            "$ref": "#/definitions/AvailableContacts"
+          },
+          "x-ms-identifiers": [],
+          "description": "A list of available contacts."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results.",
+          "readOnly": true
+        }
+      },
+      "description": "Response for the ListAvailableContacts API service call."
+    },
+    "AvailableContacts": {
+      "type": "object",
+      "properties": {
+        "spacecraft": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/ResourceReference"
+            }
+          ],
+          "description": "The reference to the spacecraft resource."
+        },
+        "groundStationName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Name of Azure Ground Station."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "allOf": [
+            {
+              "$ref": "#/definitions/ContactInstanceProperties"
+            }
+          ],
+          "description": "Properties of Contact resource."
+        }
+      },
+      "description": "Customer retrieves list of Available Contacts for a spacecraft resource. Later, one of the available contact can be selected to create a contact."
+    },
+    "AvailableGroundStationListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AvailableGroundStation"
+          },
+          "description": "A list of ground station resources."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results.",
+          "readOnly": true
+        }
+      },
+      "description": "Response for the AvailableGroundStations API service call."
+    },
+    "AvailableGroundStation": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "ID of groundStation."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Name of the ground station."
+        },
+        "location": {
+          "type": "string",
+          "description": "Azure region."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "allOf": [
+            {
+              "$ref": "#/definitions/AvailableGroundStationProperties"
+            }
+          ],
+          "description": "The properties bag for this resource."
+        }
+      },
+      "description": "Ground Stations available to schedule Contacts."
+    },
```

Comment:
Only if this is non-breaking on the wire - hopefully it is - I'd recommend the best practice of refactoring this by inheriting allOf TrackedResource from common-types.

Resource definitions - use TrackedResource or ProxyResource in commontypes
Instead of defining your own resource class with required top level properties, 'id', 'type', 'name', and 'location' and 'tags' (for TrackedResource) its recommended to inherit from ProxyResource or TrackedResource in common-types v2 or v3.json. (The v2 definition does NOT include systemData, and is otherwise similar to the v3 one).
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L56](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L56)

Reply:
it is a proxy resource. I have added allOf ProxyResource tag

Reply:
yeah we use fanout for this and that's why to broadcast it we need location